cmd/kubeadm/app/componentconfigs/defaults.go

```diff
@@ -102,7 +102,11 @@ func DefaultKubeletConfiguration(internalcfg *kubeadmapi.ClusterConfiguration) {
 	externalkubeletcfg.ReadOnlyPort = 0
 
 	// Enables client certificate rotation for the kubelet
-	externalkubeletcfg.RotateCertificates = true
+	if internalcfg.ComponentConfigs.Kubelet != nil {
```

Comment:
You can remove the else anda use
```
externalkubeletcfg.RotateCertificates = true
	if internalcfg.ComponentConfigs.Kubelet != nil {
```

Reply:
from my pov it's more explicit and readable the way I wrote it. However, if other reviewers would prefer this way I'd be happy to change the code.

Reply:
@yagonobre thank you for review!

Reply:
IMHO, although I know both are correct, I see @bart0sh 's pattern in usual, I think.